src/search/include/search/search.h

```diff
@@ -344,24 +318,89 @@ class TreeSearch
 		}
 	}
 
-	/** Get the size of the given sub-tree.
-	 * @param node The sub-tree to get the size of, defaults to the whole tree if omitted
-	 * @return The number of nodes in the sub-tree, including the node itself
+	/** Get the size of the search graph.
+	 * @return The number of nodes in the search graph
 	 */
 	size_t
-	get_size(Node *node = nullptr)
+	get_size() const
 	{
-		if (node == nullptr) {
-			node = get_root();
+		std::lock_guard lock{nodes_mutex_};
+		return nodes_.size();
+	}
+
+	/** Get the current search nodes. */
+	const std::map<std::set<CanonicalABWord<Location, ActionType>>, std::shared_ptr<Node>> &
+	get_nodes()
+	{
+		return nodes_;
+	}
+
+private:
+	std::pair<std::set<Node *>, std::set<Node *>>
+	compute_children(Node *node)
```

Comment:
Same here: Do we need to check for nullptr?

Reply:
:+1: done